File: app/src/main/java/com/galtashma/parsedashboard/Hash.java
package com.galtashma.parsedashboard;

import android.util.Log;

import java.security.MessageDigest;

public class Hash {
    private static String salt = "1m4bqk";
    public static String sha1(String value){
        try
        {
            MessageDigest messageDigest = MessageDigest.getInstance("SHA-1");
            messageDigest.update((value + salt).getBytes("UTF-8"));
            byte[] bytes = messageDigest.digest();
            StringBuilder buffer = new StringBuilder();
            for (byte b : bytes)
            {
                buffer.append(Integer.toString((b & 0xff) + 0x100, 16).substring(1));
            }
            String result = buffer.toString();
            Log.d(Const.TAG, "Hash result " + result);

            return result;
        }
        catch (Exception ignored)
        {
            ignored.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/com/galtashma/parsedashboard/Hash.kt
package com.galtashma.parsedashboard

import android.util.Log
import java.security.MessageDigest
import kotlin.experimental.and

/**
 * Created by gal on 3/16/18, rewritten by Cyb3rKo on 05/12/22.
 */

object Hash {
    private const val salt = "1m4bqk"

    fun sha1(value: String): String? {
        try {
            val messageDigest = MessageDigest.getInstance("SHA-1")
            messageDigest.update((value + salt).toByteArray(Charsets.UTF_8))
            val bytes = messageDigest.digest()
            val buffer = StringBuilder()
            bytes.forEach {
                buffer.append(((it and 0xff.toByte()) + 0x100).toString(16).drop(1))
            }
            val result = buffer.toString()
            Log.d(Const.TAG, "Hash result $result")

            return result
        } catch (ignored: Exception) {
            ignored.printStackTrace()
            return null
        }
    }
}
